lib/src/engine/compressed_texture_formats/ktx_texture.dart

```diff
@@ -0,0 +1,72 @@
+part of stagexl.engine;
+
+// https://registry.khronos.org/KTX/specs/1.0/ktxspec.v1.html
+class KtxFormat {
+  static const ktx_magic = [0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A];
+  static const ktx_end_le = 0x04030201;
+  static const ktx_end_be = 0x01020304;
+}
+
+class KtxTexture extends CompressedTexture {
+  late final int _glInternalFormat;
+  late final int _texDataOffset;
+  late final int _imageSize;
+
+  KtxTexture(super.buffer) : super() {
+    _parseHeader();
+  }
+
+  void _parseHeader() {
+    final bytes = ByteArray.fromBuffer(_buffer);
+    final magic = List.generate(12, (_) => bytes.readUnsignedByte());
+
+    for (final entry in magic.asMap().entries) {
+      if (entry.value != KtxFormat.ktx_magic[entry.key]) {
+        print('unrecognized magic, not a ktx header');
+        return;
```

Comment:
I think we should probably `throw` here.

Reply:
I thought so too, but I assumed there was a reason you made the PVR parser just print. I'll change both to throw.